control-plane/agents/src/bin/ha/cluster/main.rs

```diff
@@ -28,20 +50,44 @@ impl Cli {
     }
 }
 
+/// Once cell static variable to store the grpc client and initialize once at startup
+pub static CORE_CLIENT: OnceCell<VolumeClient> = OnceCell::new();
+
+fn initialize_tracing(args: &Cli) {
+    utils::tracing_telemetry::init_tracing(
+        "agent-ha-cluster",
+        args.tracing_tags.clone(),
+        args.jaeger.clone(),
+    )
+}
+
 #[tokio::main]
 async fn main() -> anyhow::Result<()> {
     utils::print_package_info!();
 
     let cli = Cli::args();
 
-    utils::tracing_telemetry::init_tracing(
-        "agent-ha-cluster",
-        cli.tracing_tags.clone(),
-        cli.jaeger.clone(),
-    );
+    initialize_tracing(&cli);
+
+    CORE_CLIENT
+        .set(VolumeClient::new(cli.core_grpc, None).await)
+        .ok()
+        .expect("Expect to be initialised only once");
 
-    server::ClusterAgent::new(cli.grpc_endpoint)
+    let store = etcd::EtcdStore::new(cli.store, cli.store_timeout.into()).await?;
+    let node_list = nodes::NodeList::new();
+
+    let mover = volume::VolumeMover::new(store.clone(), node_list.clone());
```

Comment:
Looks like we're passing a new to both `VolumeMover` and `ClusterAgent`, can't they just new one themselves?